### PR TITLE
New updates

### PR DESCRIPTION
**Pull request by clarisaleu**
- Added database for posts - still need to implement methods throughout
- Other ui and style things

### CODE REVIEW

**Review by jocelynshen — APPROVED** [review]

File: app/src/main/java/com/example/footprnt/Discover/Models/Business.java [hunk 1]
```diff
@@ -12,10 +19,10 @@
     private String website;
     private double rating;
     private String imageUrl;
-    private ArrayList<String> address = new ArrayList<>();
+    private ArrayList<String> address;
     private double latitude;
     private double longitude;
-    private ArrayList<String> categories = new ArrayList<>();
+    private ArrayList<String> categories;
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Discover/Models/Business.java:25]: When you modify these files change change them to the proper formatting (ie I don't need you to go back and change them but if you're touching the file change variable names to be mVariableName)

File: app/src/main/java/com/example/footprnt/Discover/Services/YelpService.java [hunk 2]
```diff
@@ -18,38 +18,30 @@
...
-
-
-        OkHttpClient client = new OkHttpClient.Builder()
-                .build();
-
-        HttpUrl.Builder urlBuilder = HttpUrl.parse(Constants.YELP_BASE_URL + location + "&term=" + query).newBuilder();
-        //urlBuilder.addQueryParameter(Constants.YELP_LOCATION_QUERY_PARAMETER, location);
-        String url = urlBuilder.build().toString();
-
+        String url = DiscoverConstants.YELP_BASE_URL + location + "&term=" + query;
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Discover/Services/YelpService.java:39]: String.format(%s%s&term=%s, DiscoverConstants.YELP_BASE_URL, location, query)

File: app/src/main/java/com/example/footprnt/Discover/Services/YelpService.java [hunk 3]
```diff
@@ -18,38 +18,30 @@
...
         Request request = new Request.Builder()
                 .url(url)
-                .header("Authorization", Constants.YELP_TOKEN)
+                .header("Authorization", DiscoverConstants.YELP_TOKEN)
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Discover/Services/YelpService.java:42]: Make "Authorization" a constant

File: app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java [hunk 4]
```diff
@@ -304,21 +306,24 @@ public void done(com.parse.ParseException e) {
...
     public static Marker createMarker(GoogleMap mMap, MarkerDetails md) throws com.parse.ParseException {
-        Post post = (Post) md.getPost();
-        double latitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLatitude();
-        double longitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLongitude();
-        String title = (post.fetchIfNeeded().getString("title"));
-        ParseFile image = post.fetchIfNeeded().getParseFile("image");
-        String imageUrl = "";
-        if (image != null) {
-            imageUrl = image.getUrl();
+        if(md.getPost()!=null) {
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java:309]: if(md.getPost() != null) {

File: app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java [hunk 5]
```diff
@@ -304,21 +306,24 @@ public void done(com.parse.ParseException e) {
...
+            Post post = (Post) md.getPost();
+            double latitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLatitude();
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java:311]: Make "location" a constant. It also might be easier here to do ->
  Latlng location = post.fetchIfNeeded().getParseGeoPoint("location");
and just use `location.getlatitude()` and `location.getLongitude()` in your code when needed

File: app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java [hunk 6]
```diff
@@ -304,21 +306,24 @@ public void done(com.parse.ParseException e) {
...
+            double longitude = (post.fetchIfNeeded().getParseGeoPoint("location")).getLongitude();
+            String title = (post.fetchIfNeeded().getString("title"));
+            ParseFile image = post.fetchIfNeeded().getParseFile("image");
+            String imageUrl = "";
+            if (image != null) {
+                imageUrl = image.getUrl();
+            }
+            Marker m = mMap.addMarker(new MarkerOptions()
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Map/Util/MapUtil.java:319]: Why not just return this directly? 
  return mMpa.addMarker(....);

File: app/src/main/java/com/example/footprnt/Util/AppUtil.java [hunk 7]
```diff
@@ -71,8 +75,20 @@ public String getAddress(Context context, LatLng point) {
         return null;
     }
 
+    /**
```
Comment (ajankenm) [inline app/src/main/java/com/example/footprnt/Util/AppUtil.java:78]: This isn't a great Javadoc =P

**Pull request merged**